sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -81,6 +81,8 @@ class Analyzer(
       UnresolvedHavingClauseAttributes ::
       HiveTypeCoercion.typeCoercionRules ++
       extendedResolutionRules : _*),
+    Batch("Cleanup", fixedPoint,
```

Comment:
I guess I was thinking we would do this in the same place that we remove subqueries.